rex/multi_year_resource.py

```diff
@@ -153,30 +196,24 @@ def _get_years(file_paths, years):
             List of integer years corresponding to the file_paths list
         """
 
+        fp_years = [int(parse_year(os.path.basename(fp), option='raise'))
+                    for fp in file_paths]
+
         if years is None:
-            years = [parse_year(os.path.basename(fp), option='raise')
-                     for fp in file_paths]
-
-        new_list = []
-        years = sorted([int(y) for y in years])
-        for year in years:
-            for fp in file_paths:
-                if str(year) in os.path.basename(fp):
-                    new_list.append(fp)
-                    break
-
-        if not new_list:
-            msg = ('No files were found for the given years:\n{}'
-                   .format(years))
-            raise RuntimeError(msg)
-
-        if len(new_list) != len(years):
-            msg = ('Found a bad file listing for requested years. '
-                   'Years requested: {}, files found: {}'
-                   .format(years, new_list))
-            raise RuntimeError(msg)
-
-        file_paths = new_list
+            years = fp_years
+        elif any(int(y) not in fp_years for y in years):
+            years = sorted([int(y) for y in years])
+            raise RuntimeError('Requested years "{}" not all found in '
+                               'file years "{}"'.format(years, fp_years))
+        else:
+            filtered_fps = []
+            years = sorted([int(y) for y in years])
+            for target_year in years:
+                for fp_year, fp in zip(fp_years, file_paths):
+                    if int(target_year) == int(fp_year):
+                        filtered_fps.append(fp)
+                        break
```

Comment:
So does this mean that you can have multiple years in one file but not one year over multiple files?

Reply:
For the multi-year handlers, yes. I'm not sure why it was originally designed this way but the multi-time handlers do multiple files within a year.

Reply:
Ah okay, gotcha. We allow so many configurations these days that it's becoming harder to keep track! I will go familiarize myself more with the multi-time handlers. Thanks!

Reply:
Yeah i'm not sure if they're ever used in our work... I think they were developed for some misc wind data michael was working with. Fine to keep em for now but at some point we could probably merge the multi-time and multi-year handlers (they seem like they should fit into a common framework).

Reply:
Right, the blurry distinction was probably why I was getting all confused. In any case, I think what you have for the time-being looks great!